src/index.ts

```diff
@@ -403,37 +400,40 @@ export class Upload extends Pumpify {
     }
 
     if (this.userProject) {
-      reqOpts.params = reqOpts.params || {};
-      reqOpts.params.userProject = this.userProject;
+      reqOpts.qs = reqOpts.qs || {};
+      reqOpts.qs.userProject = this.userProject;
     }
 
-    reqOpts.validateStatus = (status: number) => {
-      return (status >= 200 && status < 300) ||
-          status === RESUMABLE_INCOMPLETE_STATUS_CODE;
-    };
-
-    this.authClient.request(reqOpts).then(
-        r => {
-          return callback(null, r, r.data);
-        },
-        (err: AxiosError) => {
-          const body = err.response ? err.response.data : undefined;
-          const e = (body && body.error) ? body.error : err;
-          return callback(e, err.response, body);
+    this.authClient.getRequestHeaders()
+        .then(authHeaders => {
+          reqOpts.headers = Object.assign({}, reqOpts.headers, authHeaders);
+          request(reqOpts, (err, res, body) => {
+            let e = (body && body.error) ? body.error : err;
+            // If no error was returned, but the response had an invalid status
+            // code, create a new error to be passed to the callback.
+            if (!e && (res.statusCode < 200 || res.statusCode >= 300) &&
+                res.statusCode !== RESUMABLE_INCOMPLETE_STATUS_CODE) {
+              e = new Error(`The request failed with a ${res.statusCode}.`);
```

Comment:
We could pop a `code` property on here with the status code, so it's familiar for a user to sniff it out.

Reply:
Great idea.  Added!